pandas/core/indexes/period.py

```diff
@@ -350,6 +351,17 @@ def _int64index(self) -> Int64Index:
     # ------------------------------------------------------------------------
     # Index Methods
 
+    def __getitem__(self, key):
+        # PeriodArray.__getitem__ returns PeriodArray for 2D lookups,
+        #  so we need to issue deprecation warning and cast here
```

Comment:
Are you planning to split this into it's own PR? Don't want to have it held up by review here.

Reply:
hadnt planned on it, but might as well.  this PR is back-burner until at least the frame-with-frame case is done